### ISSUE TITLE
[3][firehose-fetcher] Service is keep restarting in stage/production

### ISSUE THREAD

**Issue by pkajaba** [openshiftio/openshift.io#956]
There seems to be a problem with liveness probe in firehose-fetcher service. Because of this container keeps restarting. 

There was PR created to ease problems with liveness problem (https://github.com/fabric8-analytics/fabric8-analytics-firehose-fetcher/pull/19), but it's not guaranteed that it will really fix this issue so a goal of this task would be to investigate the root cause of it.

**Comment by msrb** [comment]
@pkajaba please update this issue as we discussed yesterday 😉

**Comment by pkajaba** [comment]
As we discussed those exceptions are usually caused because of 2 things:
1) A problem in the library which we are using for reading server-side events. If there are 0 bytes read from stream exception is thrown which leads to restart of a container. Unfortunately, there is no documentation for firehose-fetcher from libraries.io but from doing some research 0 usually, means the end of the stream. which leads me to a 2nd cause.
2) libraries.io firehose-fetcher is either not running or time-outing. 

So usual "workflow" is that there is 0 bytes read exception and there is downtime for a few minutes. We agreed in a team that we could catch these exceptions and try to restart stream in a service, but we decided to let Openshift handle it.
